client/rest/src/main/java/org/elasticsearch/client/RuntimeInfo.java

```diff
@@ -0,0 +1,145 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
+class RuntimeInfo {
+
+    /**
+     * Returns runtime info to append to the metadata header. Currently looks up classes identifying non-Java JVM
+     * languages and appends their major/minor version patch.
+     */
+    public static String getRuntimeMetadata() {
+        StringBuilder s = new StringBuilder();
+        String v;
+
+        v = HlrcKind();
+        if (v != null) {
+            s.append(",hl=").append(v);
+        }
+
+        v= kotlinVersion();
+        if (v != null) {
+            s.append(",kt=").append(v);
+        }
+
+        v = scalaVersion();
+        if (v != null) {
+            s.append(",sc=").append(v);
+        }
+
+        v = groovyVersion();
+        if (v != null) {
+            s.append(",clj=").append(v);
+        }
+
+        v = groovyVersion();
+        if (v != null) {
+            s.append(",gy=").append(v);
+        }
+
+        return s.toString();
+    }
+
+    public static String HlrcKind() {
+        try {
+            Class.forName("org.elasticsearch.client.RestHighLevelClient");
+        } catch (Throwable t) {
+            return null;
+        }
+        // 1 is the HLRC based on ES server request/response classes
+        // 2 will be the HLRC based on code generation from API specs
+        return "1";
+    }
+
+    public static String kotlinVersion() {
+        try {
+            //KotlinVersion.CURRENT.toString()
+            Class<?> clazz = Class.forName("kotlin.KotlinVersion");
+            Field field = clazz.getField("CURRENT");
+            String version = field.get(null).toString();
+            return stripPatchRevision(version);
+
+        } catch (Throwable t) {
+            // ignore
+        }
+        return null;
+    }
+
+    public static String scalaVersion() {
+        try {
+            // scala.util.Properties.versionNumberString()
+            Class<?> clazz = Class.forName("scala.util.Properties");
+            Method m = clazz.getMethod("versionNumberString");
+            String version = (String) m.invoke(null);
+            return stripPatchRevision(version);
+
+        } catch (Throwable t) {
```

Comment:
I believe we should only catch `Exception` here to allow for `OOM` errors to propagate

Reply:
Good point. I wanted to avoid any kind of class-loading problem to cause returning `null` but the non-exception we can encounter (besides OOM) are things like `LinkageError` that we'd better let bubble up as well.

Changed to `Exception`.